test/server/cards/02_SHD/units/BoKatanKryzeFightingForMandalore.spec.ts

```diff
@@ -0,0 +1,81 @@
+describe('Bo-KatanKryze', function () {
+    integration(function (contextRef) {
+        describe('Bo-KatanKryze on defeated ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['bokatan-kryze#fighting-for-mandalore'],
+                        base: { card: 'echo-base', damage: 12 }
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine'],
+                        base: { card: 'echo-base', damage: 12 }
+                    }
+                });
+            });
+
+            it('No cards drawn if damage on both bases is less than 15', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.bokatanKryze);
+                context.player1.clickCard(context.battlefieldMarine);
+                expect(context.bokatanKryze).toBeInZone('discard');
+                expect(context.battlefieldMarine).toBeInZone('discard');
+
+                expect(context.player1.hand.length).toBe(0);
+                expect(context.player2).toBeActivePlayer();
+            });
+
+            it('1 card drawn if damage on opponents base is equal to 15 and own base is less than 15', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.bokatanKryze);
+                context.player1.clickCard(context.p2Base);
+                expect(context.p2Base.damage).toBe(15);
+
+                context.player2.clickCard(context.battlefieldMarine);
+                context.player2.clickCard(context.bokatanKryze);
+                expect(context.player1.hand.length).toBe(1);
+            });
+
+            it(' 2 cards drawn if damage on both bases exceeds 15', function () {
```

Comment:
We have this issue where the Jasmine tests are super heavy and slow, and we have finite Github Actions time in the free tier. So, unfortunately we have to consolidate multiple cases into one where possible.

On-attack abilities are a really good candidate for this because we can reset the attacker by setting exhausted = false in between attacks. See the Mace Windu Party Crasher tests for an example of a typical pattern we use. We could use a card like Repair to bring basr damage back below 15 if needed for a test's setup.

Reply:
No worries, Ill finish what i'm doing on another branch and make the updates to this! 
If I am reading correctly, its simply consolidate all test cases under 1 iteration to save on resources?

Reply:
@AMMayberry1 updated